source/s3_auto_ranged_put.c

```diff
@@ -502,6 +502,10 @@ static bool s_s3_auto_ranged_put_update(
                 if (auto_ranged_put->upload_id == NULL) {
                     goto no_work_remaining;
                 }
+                if (auto_ranged_put->base.synced_data.finish_result.error_code == AWS_ERROR_SUCCESS) {
+                    /* Not sending abort when success */
+                    goto no_work_remaining;
```

Comment:
I'm have a really hard time following the flow of this function...

If we get here, what is it that finished successfully?

Reply:
updated the comment. This function is truly hard to follow. Maybe we can change all those boolean about each request state to something better...